analysis/analysis_utils.py

```diff
@@ -481,19 +481,21 @@ def load_results(
             print("---------------------------------------")
 
             for i, line in enumerate(results):
+                # Stop as soon as we hit a result below our limit, because anything after
+                # that is not guarrenteed to be valid due to potential on-GPU filtering.
+                if line.lh < lh_level:
```

Comment:
I guess I should say that this PR doesn't change the global behavior. Since the results with a likelihood under lh_limit get filtered later on. This just moves up the filtering.

Reply:
yeah i agree with jeremy, same behavior as before but makes the code a bit cleaner!